### PR TITLE
[SYCL][Doc] Fix documentation link

### PR DESCRIPTION
**Pull request by gmlueck**

### CODE REVIEW

File: sycl/doc/EnvironmentVariables.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ compiler and runtime.
...
+| `ONEAPI_DEVICE_SELECTOR` | [See below.](#oneapi_device_selector)  | This device selection environment variable can be used to limit the choice of devices available when the SYCL-using application is run.  Useful for limiting devices to a certain type (like GPUs or accelerators) or backends (like Level Zero or OpenCL).  This device selection mechanism is replacing `SYCL_DEVICE_FILTER` .  The `ONEAPI_DEVICE_SELECTOR` syntax is shared with OpenMP and also allows sub-devices to be chosen. [See below.](#oneapi_device_selector) for a full description.   |
 | `SYCL_DEVICE_FILTER` (deprecated) | `backend:device_type:device_num` | Please use `ONEAPI_DEVICE_SELECTOR` environment variable instead. See section [`SYCL_DEVICE_FILTER`](#sycl-device-filter) below for `SYCL_DEVICE_FILTER` description. |
```
Comment (steffenlarsen) [inline sycl/doc/EnvironmentVariables.md:11]:
```suggestion
| `SYCL_DEVICE_FILTER` (deprecated) | `backend:device_type:device_num` | Please use `ONEAPI_DEVICE_SELECTOR` environment variable instead. See section [`SYCL_DEVICE_FILTER`](#sycl_device_filter) below for `SYCL_DEVICE_FILTER` description. |
```

File: sycl/doc/EnvironmentVariables.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ compiler and runtime.
...
 | `SYCL_DEVICE_ALLOWLIST` | See [below](#sycl-device-allowlist) | Filter out devices that do not match the pattern specified. `BackendName` accepts `host`, `opencl`, `level_zero` or `cuda`. `DeviceType` accepts `host`, `cpu`, `gpu` or `acc`. `DeviceVendorId` accepts uint32_t in hex form (`0xXYZW`). `DriverVersion`, `PlatformVersion`, `DeviceName` and `PlatformName` accept regular expression. Special characters, such as parenthesis, must be escaped. DPC++ runtime will select only those devices which satisfy provided values above and regex. More than one device can be specified using the piping symbol "\|".|
```
Comment (steffenlarsen) [inline sycl/doc/EnvironmentVariables.md:12]:
```suggestion
| `SYCL_DEVICE_ALLOWLIST` | See [below](#sycl_device_allowlist) | Filter out devices that do not match the pattern specified. `BackendName` accepts `host`, `opencl`, `level_zero` or `cuda`. `DeviceType` accepts `host`, `cpu`, `gpu` or `acc`. `DeviceVendorId` accepts uint32_t in hex form (`0xXYZW`). `DriverVersion`, `PlatformVersion`, `DeviceName` and `PlatformName` accept regular expression. Special characters, such as parenthesis, must be escaped. DPC++ runtime will select only those devices which satisfy provided values above and regex. More than one device can be specified using the piping symbol "\|".|
```

**Review by steffenlarsen — COMMENTED** [review]
It seems like the two other links are broken as well. I assume they need the kind of fix, so we might as well do it while we are in here. 😄

File: sycl/doc/EnvironmentVariables.md (hunk 1 above)
Comment (gmlueck) [reply sycl/doc/EnvironmentVariables.md]: I fixed this an some others in 7def906d2644f72d1f381a045334797ed0559049.

File: sycl/doc/EnvironmentVariables.md (hunk 2 above)
Comment (gmlueck) [reply sycl/doc/EnvironmentVariables.md]: I fixed this an some others in 7def906d2644f72d1f381a045334797ed0559049.

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Pull request merged**